### PR TITLE
XYChart: Use same auto-ranging logic for X and Y axes

### LINKED ISSUE #78616: XY Chart Adjust X axis max
### What happened?

Dear grafana community,

on the XY-Chart the limits do not work correctly.

If Min and Max are set individually for the x AND y axis (overrides), it only works for the y axis. In my case the x-axis should also be limited.

### What did you expect to happen?

The overrides on the y-axis work, the max and min are set properly there, but the x axis completely ignores the settings.

### Did this work before?

No

### How do we reproduce it?

N/A

### Is the bug inside a dashboard panel?

_No response_

### Environment (with versions)?

Grafana:
OS:
Browser:

### Grafana platform?

None

### Datasource(s)?

_No response_

### PR DESCRIPTION
**Pull request by ryanjpwatts**
**What is this feature?**

Removes the early return done for the x-axis in the UPlotScaleBuilder. This will enable users to use soft min/max on all axes. Modifying the soft min/max on the Axis settings will update all axes, and making use of the override tab will allow a user to set a soft min/max on a specific axis.

**Why do we need this feature?**

We need the ability to be able to set soft min/max limits on all axes, not just the y-axis, which is the issue I'm running into when using the XY Chart.

**Who is this feature for?**

Dashboard creators.

**Which issue(s) does this PR fix?**:

Fixes #78616

### CODE REVIEW

**Comment by leeoniya** [comment]
hey @ryanjpwatts thanks for the PR!

i think min/max config for x axis is useful in both XY and Trend panels. i actually expected this change to affect Trend and not XY, but it does the opposite :) upon further digging it turns out we set a custom range function for Trend, with much simplified logic that only handles hard limits but not soft limits:

https://github.com/grafana/grafana/blob/eeabb6f066ae96f88239fe00e7f26b0ccfafe26c/public/app/core/components/TimeSeries/utils.ts#L177-L182

whereas in XY we simply use the default `range` behavior:

https://github.com/grafana/grafana/blob/eeabb6f066ae96f88239fe00e7f26b0ccfafe26c/public/app/plugins/panel/xychart/v2/scatter.ts#L364-L377

this PR does change the default XY behavior, which will now get the same 10% scale padding to each side along X as it does along Y. that seems like an okay change.

File: docs/sources/panels-visualizations/visualizations/bar-chart/index.md [hunk 1]
```diff
@@ -208,7 +208,7 @@ By setting the width of the axis, data whose axes types are different can share
 
 ### Soft min and soft max
 
-Set a **Soft min** or **soft max** option for better control of Y-axis limits. By default, Grafana sets the range for the Y-axis automatically based on the dataset.
+Set a **Soft min** or **soft max** option for better control of all axis limits. By default, Grafana sets the range for the Y-axis automatically based on the dataset.
```
Comment (leeoniya) [inline docs/sources/panels-visualizations/visualizations/bar-chart/index.md:211]: this does not apply to BarChart

Comment (ryanjpwatts) [reply docs/sources/panels-visualizations/visualizations/bar-chart/index.md]: Fixed in latest commit, accidentally updated the wrong one, apologies!

**Review by leeoniya — APPROVED** [review]

**Comment by IdjazB** [comment]
@ryanjpwatts @leeoniya whats the status on the above? are we able to adjust the x-axis of XY Chart using the max/min overrides? Seems it still only adjusts the y-axis. Any advice would be much appreciated, thank you

**Comment by ryanjpwatts** [comment]
> whats the status on the above? are we able to adjust the x-axis of XY Chart using the max/min overrides? Seems it still only adjusts the y-axis. Any advice would be much appreciated, thank you

@IdjazB working fine on the latest version of Grafana Cloud.

https://github.com/grafana/grafana/assets/33863267/996a6fda-724f-4165-b576-c61ad85b1a81

**Pull request merged**